api/k8s/v1/k8s.proto

```diff
@@ -113,8 +113,10 @@ message Pod {
   map<string, string> annotations = 10;
 }
 
-message ListPodsOptions {
-  map<string, string> labels = 9;
+// List Options are suitable for all resources types eg: pods, HPA, and deployment.
+message ListOptions {
+  map<string, string> labels = 1;
+  string field_selectors = 2 [ (validate.rules).string = {min_bytes : 1} ];
```

Comment:
can you add individual docstrings for these fields? particularly field_selectors.

it's also a bit weird to me that we have labels instead of label_selectors. or should we eventually have both?

finally wondering field_selectors should be a repeated string where the comma separated join for multiple selectors is done on backend

Reply:
The `labels` field existed before this pr, but yes I can refactor them to be `label_selectors` as thats what it translates into.

I had mix feelings on `field_selectors` being a string vs repeated, as they support a verity of [operators](https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/) such as `!= | ==` I could support this of course, but in the moment it felt fine to just let the caller construct their own query string based on k8s docs and will just pass this through to the `ListOptions` untouched.

If you feel strongly about the more structured approach let me know and I can do so.

Reply:
Talked offline, going to revert these proto changes for now as its not longer being used in this PR. Going to make a follow up PR to enable field selector functionality with additional List functionality for all k8s resources.